src/RadioButton.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { color } from 'styled-system'
+import Icon from './Icon'
+import Text from './Text'
+
+const RadioButton = ({
+  className,
+  theme,
+  onClick,
+  isSelected,
+  disabled,
+  radioButtonText
+}) => {
+  const radioIconName = isSelected ? 'radioFilled' : 'radio'
+  const radioIconColor = isSelected ? 'blue' : 'borderGray'
+
+  function handleClick() {
+    if (disabled || isSelected) return
```

Comment:
Using the native `disabled` and `checked` attributes and React's built-in `onChange` handler prevents the event from being called by default, so this can be cut